src/dialog.scss

```diff
@@ -47,6 +47,7 @@ $button: #{$fd-namespace}-button;
   @extend %dialog;
 
   display: none;
+  z-index: 999;
```

Comment:
not sure if we are safe to hardcode z-index.
we already have [this mixing](https://github.com/SAP/fundamental-styles/blob/7c93b2c83dcedbd68b5c189f35fdefe787edb32b/src/_new-settings.scss#L16).
@InnaAtanasova please advise the approach to use.

Reply:
We already have something similar with [popover](https://github.com/SAP/fundamental-styles/blob/7c93b2c83dcedbd68b5c189f35fdefe787edb32b/src/popover.scss#L10)

Reply:
We avoid adding z-index unless it's really necessary and then we avoid using numbers like 999 and 1001. When our lib is used in apps with other UI components the z-indexes introduce "bugs".

Reply:
It's made to have ability to have overlays over the dialogs, [some more information](https://github.com/SAP/fundamental-styles/pull/2390#issuecomment-850813451).

Reply:
I would introduce variable $dialog-initial-z-index: 999, and use it for incrementing the necessary z-index, depending on the element itself, at least this will give us more control in the future